Generators/src/GeneratorFromFile.cxx

```diff
@@ -0,0 +1,78 @@
+#include "Generators/GeneratorFromFile.h"
+#include <TTree.h>
+#include <TFile.h>
+#include <TBranch.h>
+#include <FairLogger.h>
+#include <TParticle.h>
+#include <TClonesArray.h>
+#include <FairPrimaryGenerator.h>
+#include <cstdio>
+
+ClassImp(o2::eventgen::GeneratorFromFile)
+
+namespace o2 {
+namespace eventgen {
+
+  GeneratorFromFile::GeneratorFromFile(const char *name) {
+    mEventFile = TFile::Open(name);
+    if (mEventFile == nullptr) {
+      LOG(ERROR) << "EventFile " << name << " not found \n"; 
+      return;
+    }
+    // the kinematics will be stored inside a Tree "TreeK" with branch "Particles"
+    // different events are stored inside TDirectories
+
+    // we need to probe for the number of events
+    char eventstring[7];
+    TObject *object = nullptr;
+    do {
+      sprintf(eventstring, "Event%d", mEventsAvailable);
+      //std::cout << "probing for " << eventstring << "\n";
+      object = mEventFile->Get(eventstring);
+      //std::cout << "got " << object << "\n";
+      if (object != nullptr) mEventsAvailable++;
+    }
+    while (object != nullptr);
+    std::cout << "Found " << mEventsAvailable << " events in this file \n";
+  }
+  
+  Bool_t GeneratorFromFile::ReadEvent(FairPrimaryGenerator* primGen) {
+    if (mEventCounter < mEventsAvailable) {
+	// get the tree and the branch
+	char treestring[14];
```

Comment:
that limits the event count to 99, then the string is longer than the field. Use `stringstream` instead?